x/transfermiddleware/ibc_middleware.go

```diff
@@ -127,39 +127,23 @@ func (im IBCMiddleware) OnTimeoutPacket(ctx sdk.Context, packet channeltypes.Pac
 		return err
 	}
 
-	denom := data.Denom
-	coin := im.keeper.IbcTransfermiddleware.GetCoin(ctx, packet.SourceChannel, denom)
-	сhannelFeeAddress := im.keeper.IbcTransfermiddleware.GetChannelFeeAddress(ctx, packet.SourceChannel)
-	if coin != nil {
-		amount := data.Amount
-		transferAmount, ok := sdk.NewIntFromString(amount)
-		if !ok {
-			return errors.Wrapf(transfertypes.ErrInvalidAmount, "unable to parse transfer amount: %s", amount)
+	fee, found := im.keeper.IbcTransfermiddleware.GetSequenceFee(ctx, packet.Sequence)
+	if found {
+		сhannelFeeAddress := im.keeper.IbcTransfermiddleware.GetChannelFeeAddress(ctx, packet.SourceChannel)
+		if сhannelFeeAddress == "" {
+			return nil
 		}
 
-		// calculate the percentage charge
-		/*
-			coin.Percentage is 100 that means that we charge 1/100 of the transfer amount
-			coin.MinFee.Amount is the minimum fee that we charge
-			the new amount is the transfer amount minus the percentage charge and the minimum fee
-			transfer amount is a 99/100 of the original amount
-			so to get the fee we charge transferAmount.QuoRaw(coin.Percentage - 1) + coin.MinFee.Amount
-		*/
-
-		percentageCharge := sdk.NewInt(0)
-		if coin.Percentage > 1 {
-			percentageCharge = transferAmount.QuoRaw(coin.Percentage - 1)
-		}
-		percentageCharge = percentageCharge.Add(coin.MinFee.Amount)
-
 		fee_address, err := sdk.AccAddressFromBech32(сhannelFeeAddress)
 		if err != nil {
-			return errors.Wrapf(err, "failed to decode receiver address: %s", сhannelFeeAddress)
+			return nil
+			// return errors.Wrapf(err, "failed to decode receiver address: %s", сhannelFeeAddress)
```

Comment:
Why we stop returning error here?

Reply:
if not enough coins on fee account or account address is incorrect then we just not return fee but return all the amount to not freeze all refund amount until fee address has enough or address of fee account is correct.

Reply:
Ok, then remove the commented code if it's node needed anymore plese

Reply:
removed.